run.py

```diff
@@ -240,140 +259,114 @@ def main():
-
-            sess.execute(sql_command)
-            sess.commit()
-
-        sess.close()
+        sql_sync = nonbib.NonBib(args.rowViewSchemaName, config)
+        sql_sync.create_joined_rows(nonbib_db_conn)
 
     elif args.command == 'createMetricsTable' and args.metricsSchemaName:
         m = metrics.Metrics(args.metricsSchemaName)
-        m.create_metrics_table()
+        m.create_metrics_table(metrics_db_engine)
+
     elif args.command == 'dropMetricsTable' and args.metricsSchemaName:
         m = metrics.Metrics(args.metricsSchemaName)
-        m.drop_metrics_table()
+        m.drop_metrics_table(metrics_db_engine)
+
     elif args.command == 'populateMetricsTable' and args.rowViewSchemaName and args.metricsSchemaName:
-        # m = metrics.Metrics(None, {'FROM_SCRATCH': False, 'COPY_FROM_PROGRAM': False})
-        m = metrics.Metrics(None, {'FROM_SCRATCH': True, 'COPY_FROM_PROGRAM': False})
-        m.update_metrics_all(args.rowViewSchemaName)
+        m = metrics.Metrics()
+        m.update_metrics_all(metrics_db_conn, nonbib_db_conn, args.rowViewSchemaName)
+
     elif args.command == 'populateMetricsTableDelta' and args.rowViewSchemaName and args.metricsSchemaName:
         m = metrics.Metrics(args.metricsSchemaName, {'FROM_SCRATCH': False, 'COPY_FROM_PROGRAM': False})
         m.update_metrics_changed(args.rowViewSchemaName)
-    elif args.command == 'populateMetricsTableMeta' and args.rowViewSchemaName and args.metricsSchemaName:
-        # run copy from program command
-        # that sql command will populate metrics table
-        # create metrics to get a database connection for session
-        m = metrics.Metrics(args.metricsSchemaName, {'COPY_FROM_PROGRAM': False})
-        Session = sessionmaker()
-        sess = Session(bind=m.connection)
-
-        command_args = ' populateMetricsTable '  \
-            + ' --rowViewSchemaName ' + args.rowViewSchemaName \
-            + ' --metricsSchemaName ' + args.metricsSchemaName
-        python_command = "'python " + os.path.abspath(__file__) + command_args + "'"
-        sql_command = 'copy ' + args.metricsSchemaName + '.metrics'   \
-            + ' from program ' + python_command + ';'
-
-        sess.execute(sql_command)
-        sess.commit()
-        sess.close()
+
     elif args.command == 'renameSchema' and args.rowViewSchemaName and args.rowViewBaselineSchemaName:
-        sql_sync = row_view.SqlSync(args.rowViewSchemaName, config)
-        sql_sync.rename_schema(args.rowViewBaselineSchemaName)
+        sql_sync = nonbib.NonBib(args.rowViewSchemaName)
+        sql_sync.rename_schema(nonbib_db_conn, args.rowViewBaselineSchemaName)
+
     elif args.command == 'createDeltaRows' and args.rowViewSchemaName and args.rowViewBaselineSchemaName:
-        sql_sync = row_view.SqlSync(args.rowViewSchemaName, config)
-        sql_sync.create_delta_rows(args.rowViewBaselineSchemaName)
+        sql_sync = nonbib.NonBib(args.rowViewSchemaName)
```

Comment:
the sql_sync instance should be created before the block of if statements

Reply:
fixed in new commit: https://github.com/spacemansteve/AdsDataSqlSync/commit/3031125f3156da97f2176dc67af5734abb378660